src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_account_scenarios.py

```diff
@@ -603,6 +603,16 @@ def test_renew_account_kerb_key(self, resource_group):
     @ResourceGroupPreparer()
     @StorageAccountPreparer()
     def test_create_account_sas(self, storage_account):
+        from azure.cli.core.azclierror import RequiredArgumentMissingError
+        with self.assertRaises(RequiredArgumentMissingError):
+            self.cmd('storage account generate-sas --resource-types o --services b --expiry 2000-01-01 '
+                     '--permissions r --account-name ""')
+
+        invalid_connection_string = "DefaultEndpointsProtocol=https;EndpointSuffix=core.windows.net;"
+        with self.assertRaises(RequiredArgumentMissingError):
+            self.cmd('storage account generate-sas --resource-types o --services b --expiry 2000-01-01 '
+                     '--permissions r --connection-string {}'.format(invalid_connection_string))
+
```

Comment:
It's existing test just below

Reply:
actually below one is using --account-key, which is default behavior when there is no credential provided.
I have such concern because I would be surprised why our test doesn't detect the issue when add none validator. So it would be better to add one here.

Reply:
My mistake. I added a `LiveScenarioTest` for `storage account generate-sas` with `--connection-string`.

Our tests covered `storage file/blob/queue generate-sas` with `--connection-string` but missed `storge account generate-sas` before😂